overlord/configstate/configmgr.go

```diff
@@ -44,6 +46,16 @@ func MockConfigcoreRun(f func(sysconfig.Device, configcore.RunTransaction) error
 func Init(st *state.State, hookManager *hookstate.HookManager) error {
 	delayedCrossMgrInit()
 
+	// Retrieve home directories
+	st.Lock()
+	defer st.Unlock()
+	tr := config.NewTransaction(st)
+	var homedirs string
+	if err := tr.GetMaybe("core", "homedirs", &homedirs); err != nil {
+		return err
+	}
+	dirs.SetSnapHomeDirs(homedirs)
+
```

Comment:
this also needs to be checked in the unit tests

Reply:
please check https://github.com/snapcore/snapd/pull/13145#issuecomment-1888070174